3b.nextflow_production/scripts/build_duckdb_views.py

```
@@ -0,0 +1,265 @@
+                {_compartment_projection(conn, "profiles", "cell_profiles", "c", "Cell", drop_object_id=True)},
+                {_compartment_projection(conn, "profiles", "cytoplasm_profiles", "cy", "Cytoplasm", drop_object_id=True)}
+            FROM images.image_assets a
+            JOIN profiles.nuclei_profiles n
+                ON n.Metadata_Imaging_ImageID = a.Metadata_Imaging_ImageID
+            JOIN profiles.cell_profiles c
+                ON c.Metadata_Imaging_ImageID = n.Metadata_Imaging_ImageID
+                AND c.Metadata_Object_ObjectID = n.Metadata_Object_ObjectID
+            JOIN profiles.cytoplasm_profiles cy
+                ON cy.Metadata_Imaging_ImageID = n.Metadata_Imaging_ImageID
+                AND cy.Metadata_Object_ObjectID = n.Metadata_Object_ObjectID
+            """
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)